### PR TITLE
Add View Command

### LINKED ISSUE #54: Implement the View Command
As a TA, I want to easily view all student contact details, to keep the toolkit current.

### PR DESCRIPTION
**Pull request by YuSoonZ**
Fix #54

### CODE REVIEW

**Review by kailashgautham — CHANGES_REQUESTED** [review]
LGTM! Have commented some minor changes / questions that I have.

File: src/test/java/seedu/address/logic/commands/AddCommandTest.java [hunk 1]
```diff
@@ -123,6 +123,11 @@ public void addPerson(Person person) {
             throw new AssertionError("This method should not be called.");
         }
 
+        @Override
+        public Person viewPerson(Person target) {
+            return null;
```
Comment (g-tejas) [inline src/test/java/seedu/address/logic/commands/AddCommandTest.java:128]: Agreed with Kailash

File: src/main/java/seedu/address/model/ModelManager.java [hunk 2]
```diff
@@ -109,6 +109,11 @@ public void addPersonKeepFilter(Person person) {
         addressBook.addPerson(person);
     }
 
+    @Override
+    public Person viewPerson(Person target) {
+        return target;
+    }
```
Comment (g-tejas) [inline src/main/java/seedu/address/model/ModelManager.java:115]: Sorry, what does this method do exactly? Seems to just return the parameter, is this necessary?

File: src/main/java/seedu/address/logic/parser/ViewCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.ViewCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new ViewCommand object.
+ */
+public class ViewCommandParser implements Parser<ViewCommand> {
+    public static final String MESSAGE_ONE_BASED_INDEXING =
+            "Please input a 1-based integer.";
```
Comment (g-tejas) [inline src/main/java/seedu/address/logic/parser/ViewCommandParser.java:15]: Do you mean index?

File: src/test/java/seedu/address/logic/parser/ViewCommandParserTest.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.ViewCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+public class ViewCommandParserTest {
+
+    private final ViewCommandParser parser = new ViewCommandParser();
```
Comment (g-tejas) [inline src/test/java/seedu/address/logic/parser/ViewCommandParserTest.java:14]: Could this initialisation be placed within a `@BeforeEach` `setUp` method?

File: src/test/java/seedu/address/logic/commands/ViewCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.Person;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for
+ * {@code DeleteCommand}.
+ */
+public class ViewCommandTest {
+
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
```
Comment (g-tejas) [inline src/test/java/seedu/address/logic/commands/ViewCommandTest.java:30]: Can the initialisation be placed in a @BeforeEach setUp call? This is so the initialisation happens once for each test, otherwise might run into bugs

File: src/test/java/seedu/address/logic/commands/ViewCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+        assertNotEquals(null, viewFirstCommand);
+
+        // different person -> returns false
+        assertNotEquals(viewFirstCommand, viewSecondCommand);
+    }
+
+    @Test
+    public void execute_emptyAddressBook_throwsCommandException() {
+        // Empty the address book
+        model = new ModelManager();
```
Comment (g-tejas) [inline src/test/java/seedu/address/logic/commands/ViewCommandTest.java:92]: If you end up doing the `setUp` method call, can check if you still need a plain ModelManager constructor with no params passed or need to pass in the UserPrefs and all

File: src/test/java/seedu/address/logic/commands/ViewCommandTest.java [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+    /**
+     * Updates {@code model}'s filtered list to show no one.
+     */
+    private void showNoPerson(Model model) {
+        model.updateFilteredPersonList(p -> false);
+
+        assertTrue(model.getFilteredPersonList().isEmpty());
+    }
```
Comment (g-tejas) [inline src/test/java/seedu/address/logic/commands/ViewCommandTest.java:125]: Is this test relevant to View class? Since its not calling any view related methods, nor is it used anywhere. Its also not marked with `@Test`

**Review by g-tejas — CHANGES_REQUESTED** [review]
Some small issues and some questions as well

File: docs/UserGuide.md [hunk 8]
```diff
@@ -125,6 +125,19 @@ Examples:
...
+View the specified person from the TA Toolkit.
+
+Format: `view INDEX`
+
+* View the person at the specified `INDEX`.
+* The index refers to the index number shown in the displayed person list.
+* The index **must be a positive integer** 1, 2, 3, …​
+
+Examples:
+* `ls` followed by `view 2` views the 2nd person in the TA Toolkit.
```
[the remark this answers is not in the document]
Comment (YuSoonZ) [reply docs/UserGuide.md:139]: Yeap, as what Tejas mentioned, its to use ls command to bring the page back to show all students within TA Toolkit, then to use view function

File: src/test/java/seedu/address/logic/commands/AddCommandTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (YuSoonZ) [reply src/test/java/seedu/address/logic/commands/AddCommandTest.java:128]: Understood, have corrected it. Thanks for spotting it!

File: docs/UserGuide.md [hunk 9]
```diff
@@ -136,7 +149,7 @@ Format: `dc INDEX`
 * The index **must be a positive integer** 1, 2, 3, …​
 
 Examples:
-* `list` followed by `dc 2` deletes the 2nd person in the TA Toolkit.
+* `ls` followed by `dc 2` deletes the 2nd person in the TA Toolkit.
```
[the remark this answers is not in the document]
Comment (YuSoonZ) [reply docs/UserGuide.md:152]: Would follow the same explanation as above

File: src/test/java/seedu/address/logic/commands/AddCommandTest.java (hunk 1 above)
Comment (YuSoonZ) [reply src/test/java/seedu/address/logic/commands/AddCommandTest.java:128]: Have updated it! Thanks

File: src/main/java/seedu/address/model/ModelManager.java (hunk 2 above)
Comment (YuSoonZ) [reply src/main/java/seedu/address/model/ModelManager.java:115]: Its due to the Model interface I put a viewPerson method within it, so the ModelManager has to implement it else there would be an error

File: src/main/java/seedu/address/logic/parser/ViewCommandParser.java (hunk 3 above)
Comment (YuSoonZ) [reply src/main/java/seedu/address/logic/parser/ViewCommandParser.java:15]: Yes, thanks for spotting it, Have amended it

File: src/test/java/seedu/address/logic/parser/ViewCommandParserTest.java (hunk 4 above)
Comment (YuSoonZ) [reply src/test/java/seedu/address/logic/parser/ViewCommandParserTest.java:14]: image
Would it be something like this?

File: src/test/java/seedu/address/logic/commands/ViewCommandTest.java (hunk 5 above)
Comment (YuSoonZ) [reply src/test/java/seedu/address/logic/commands/ViewCommandTest.java:30]: image

Have updated it as well here

File: src/main/java/seedu/address/logic/commands/ViewCommand.java [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToView = lastShownList.get(targetIndex.getZeroBased());
+        model.viewPerson(personToView);
```
[the remark this answers is not in the document]
Comment (teojunda) [reply src/main/java/seedu/address/logic/commands/ViewCommand.java:54]: Then can also remove all mentions of `viewPerson` in model and modelManager, etc.

[the remark this answers is not in the document]
Comment (YuSoonZ) [reply src/main/java/seedu/address/logic/commands/ViewCommand.java]: Have looked into it and updated it

**Review by g-tejas — APPROVED** [review]
LGTM!

**Review by kailashgautham — APPROVED** [review]
LGTM!
